src/project_types/sequence.rs

```diff
@@ -25,42 +30,180 @@ impl Sequence {
         num_sections: Option<u32>
     ) -> Result<Sequence, Error> {
         // Defaults
+        // TODO: put constraint (>25ms)
         let frame_dur_ms = frame_duration_ms.unwrap_or(50);
+        // TODO: put constraint (!= 0)
         let num_sects = num_sections.unwrap_or(1);
 
         // Create sequence
-        let sequence = Sequence {
+        let mut sequence = Sequence {
             name: name.to_string(),
             directory_name: seq_directory_name.to_string(),
             music_file_name: music_file_name.to_string(),
             music_duration_sec: music_duration_sec,
             frame_duration_ms: frame_dur_ms,
-            num_sections: num_sects,
+            num_sections: 0 // Updated in resection
         };
 
         // Section sequence
-        try!(sequence.resection());
+        try!(sequence.resection(num_sects));
 
         Ok(sequence)
     }
 
-    /// Resection a sequence
-    pub fn resection(&self) -> Result<(), Error> {
-        if self.num_sections == 1 {
-            let num_frames_f32: f32 = (self.music_duration_sec as f32 * 1000_f32) / self.frame_duration_ms as f32;
-            let num_frames = num_frames_f32.ceil() as u32;
-            let num_channels = 1; // TODO: change when add layout
-            let sequence_section = SequenceSection {
-                seq_name: self.name.to_string(),
-                index: 1,
-                num_frames: num_frames,
-                editor: None,
-                data: vec![vec![0; num_frames as usize]; num_channels],
-            };
-            sequence_section.write_to_file(&self.directory_name)
+    /// Reads in a sequence section from its file
+    pub fn get_section(&self, section: u32) -> Result<SequenceSection, Error> {
+        if self.section_in_range(section) {
+            let section_path = self.get_section_path(section);
+            utils::read_sequence_section(&section_path)
         } else {
-            Err(Error::TodoErr)
+            Err(Error::InvalidSequenceSection(section))
+        }
+    }
+
+    /// Checks to see if the given section is valid/in range
+    pub fn section_in_range(&self, section: u32) -> bool {
+        section < self.num_sections
+    }
+
+    /// Reads in all sequence sections from their files into a vec
```

Comment:
```
fn get_all_sections(&self) -> Result<Vec<SequenceSection>, Error> {
     (0..self.num_sections).map(|i| self.get_section(i)).collect()
}
```

See https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect and https://doc.rust-lang.org/std/iter/trait.FromIterator.html if you're curious.